### ISSUE TITLE
The URL mentioned in default.nix when Nix is too old is broken

### ISSUE THREAD

**Issue by copumpkin** [NixOS/nixpkgs#15344]
https://github.com/NixOS/nixpkgs/blob/master/default.nix#L5 mentions https://nixos.org/wiki/How_to_update_when_Nix_is_too_old_to_evaluate_Nixpkgs, which points people to Hydra URLs that no longer work.

I remember a thread about improving that mechanism, but even if we don't do something fancy, we should have the URLs actually work. Now that the wiki is read-only, I'm not sure how best to fix it.

**Comment by domenkozar** [comment]
I think we should just move it to the manuals.

**Comment by domenkozar** [comment]
This is really important, I'll put it on my high-pri todo.

**Comment by domenkozar** [comment]
Latest release .nixpkgs are 404-ing at https://hydra.nixos.org/release/nix/nix-1.11.2

cc @edolstra

**Comment by vcunat** [comment]
We had an issue already: https://github.com/NixOS/nixpkgs/issues/13186.

**Comment by vcunat** [comment]
And I suppose *Nix* manual is the best place.

**Comment by domenkozar** [comment]
Releases are now at http://nixos.org/releases/nix/, but there is no `.nixpkg`

**Comment by ttuegel** [comment]
What is being done to provide `.nixpkg` files? What happened to them in the first place? Is it some kind of joke that Hydra still has links for these files?

Being unable to update old Nix-based systems quickly is completely unacceptable and reflects poor attention to infrastructure.

**Comment by domenkozar** [comment]
@ttuegel I'll dig into this tomorrow.

**Comment by domenkozar** [comment]
Btw for now best workaround is:

`nix-env -i /nix/store/356dj8w9171dajl0wq32m41dhg2ngq35-nix-1.11.2`

**Comment by ttuegel** [comment]
Thank you! I have to say, I'm quite surprised that worked. One possible resolution would be to update the manual to include instructions for finding the current hash from Hydra and following this procedure. The disadvantage of the `curl`-based procedure is that its failures are... opaque.

**Comment by ttuegel** [comment]
Actually, that didn't work; now it fails with `error: unknown hash algorithm 'sha512'`. I have also tried `nix-1.12` with no success.

**Comment by domenkozar** [comment]
@ttuegel did you use the same user? It is installed into your personal profile.

**Comment by domenkozar** [comment]
@ttuegel this is not NixOS right?

**Comment by ttuegel** [comment]
@domenkozar This is NixOS. It is installed into the system profile and `nix-env --version` reports version 1.12.

**Comment by domenkozar** [comment]
@ttuegel `1.12`? That means you're using unstable Nix. Could it be that sha512 support is just broken?

**Comment by ttuegel** [comment]
@domenkozar I don't *think* so. I've tried with `1.11` and `1.12` on this machine. The workaround is to disable Firefox from my configuration; now it seems to be working. But, I use the same configuration on a different machine without any problems, so it's more subtle than sha512 simply being broken...

**Comment by vcunat** [comment]
I think your nix-daemon also needs to be new enough.

**Comment by maximvl** [comment]
I got error message on old system with Nix, went to wiki page, tried it's commands, they failed, installed Nix from scratch, then found this thread. This experience is really confusing for me as a user, please do something with misleading message.

**Comment by domenkozar** [comment]
@maximvl did following help:

```
nix-env -i /nix/store/356dj8w9171dajl0wq32m41dhg2ngq35-nix-1.11.2
```

**Comment by domenkozar** [comment]
I'm sorry it took so long, but see https://github.com/NixOS/nixpkgs/pull/17631

**Comment by matthewbauer** [comment]
Can this be closed? also for issue #23013.

**Comment by dezgeg** [comment]
Fixed in 2e91b2d585afdb3b076b3ad655579774f100bad5.
